Configcreator

gordon-zhao opened the pull request.

DerekChan65535 committed:
Merge branch 'develop'

gordon-zhao committed:
Package Creator v0.0.0.0.1

>_< The very first version of package creator

Usage: Edit the example.json, fill out the instanceName and other
information; Put the program and example.json into the same folder with
.minecraft folder, then run it!

Exists problem: In some further version of HMCL launcher, it no longer
provides the forge jar file, instead it created a json with Jar link to
official jar file, which may not work to this program jet; Only support
to zip mods, config. assets. libraries, others, scripts. Will support
more later.

Next version:
1. Add update function.
2. Fix the problem that wont list out version.
3. May provide gui interface.

gordon-zhao reviewed (commented):
Exist problems listed in commit describe :)

gordon-zhao committed:
Merge remote-tracking branch 'refs/remotes/DeckerCHAN/develop' into ConfigCreator

DerekChan65535 reviewed (changes requested):
Type: official jar file, which may not work to this program **jet**;

DeckerCHAN commented on ConfigCreator/.vs/config/applicationhost.config:
```diff
@@ -0,0 +1,1030 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
```
What is the purpose of this file? Is this necessary or simply ignore it?

gordon-zhao replied:
You can ignore this file. It is setting for visual studio

DeckerCHAN commented on ConfigCreator/.vs/config/applicationhost.config:
```diff
@@ -0,0 +1,1030 @@
+
+        <webLimits />
+
+    </system.applicationHost>
```
Seems like it contains a configuration of host a IIS server with axps.

DeckerCHAN commented on ConfigCreator/ConfigCreator/ConfigCreator.py:
```diff
@@ -0,0 +1,544 @@
+    threadLock=threading.Lock()
+
+    '''
+    class zip_file_thread (threading.Thread):
```
Remove the comment as is not necessary.

gordon-zhao replied:
It is just the problem code. I will use it after i solve the problem passing arguments to run()

gordon-zhao committed:
Changes:
-Stick into one solution

Fix:
-Versions with only a json file are able to be detected now.

gordon-zhao committed:
Change zipping() instance initial method of file_process() and folder_process()

gordon-zhao committed:
Add a lazy option list.

gordon-zhao committed:
Added compile version in dist folder

gordon-zhao commented:
Here we go!

gordon-zhao commented on ConfigCreator/ConfigCreator/ConfigCreator.py:
```diff
@@ -0,0 +1,512 @@
+                        elif not Jar_file_found and not Json_file_found:
+                            sys.stderr.write('{} is missing important file: Jar, Config.'.format(different_versions_folder))
+                            continue
+                        if not Jar_file_found:
```
Here, I change the way analyzing the version list~ The crazy forge won't be kicked now~

gordon-zhao commented on ConfigCreator/ConfigCreator/ConfigCreator.py:
```diff
@@ -0,0 +1,512 @@
+
+    # Config file
+    folder_list['config']=Dot_Minecraft_Path+'/config'
+
```
I leave a place for further function: select addition folder to zip.

gordon-zhao commented on ConfigCreator/ConfigCreator/ConfigCreator.py:
```diff
@@ -0,0 +1,512 @@
+
+    objects={}
+    threadID=1
+
```
No more limitation on the amount of objects.

gordon-zhao commented on ConfigCreator/ConfigCreator/ConfigCreator.py:
```diff
@@ -0,0 +1,512 @@
+    if not os.path.isdir(upload_path):
+        os.mkdir(upload_path)
+    dict_a=Minecraft_directory_search(Dot_Minecraft_Path)
+
```
Add a select loop, so user don't need to enter the whole version name

gordon-zhao commented on ConfigCreator/ConfigCreator/Example.json:
```diff
@@ -0,0 +1,44 @@
+  "updatePath": "",
+  "version": "V1.0",
+  "generation": 2,
+  "instanceName": "",
```
Warning: the instanceName should be filled out, otherwise it will lead to WindowError, because the file name of the generated json file is based on this. Not tested if Non-ASCII letter can pass.

gordon-zhao reviewed (commented):
Fix the problem with the version has no jar file in the same folder

gordon-zhao committed:
Add upload modules

The package contain structure of ftp and upyun upload method

gordon-zhao committed:
Support upload after packing

gordon-zhao committed:
Fix some silly problem & compile

DeckerCHAN commented:
Doing review.

DerekChan65535 reviewed (commented):
Add git ignore to skip binary files.

DeckerCHAN commented:
Moving to independent repo.
